include/support.h

```diff
@@ -53,6 +53,15 @@
 // is in-range of a char and returns it as such.
 char int_to_char(int val);
 
+constexpr bool char_is_negative([[maybe_unused]] char c)
+{
+#if (CHAR_MIN < 0) // char is signed
+	return c < 0;
+#else // char is unsigned
+	return false;
+#endif
+}
+
```

Comment:
I originally used [std::is_unsigned](https://en.cppreference.com/w/cpp/types/is_unsigned) from `<type_traits>`:

``` c++
  return std::is_unsigned<char>::value ? false : char < 0;
```

But if `char` really is unsigned, then the `char < 0` will never be evaluated and the compiler will warn about that second condition never being evaluated.